raft/server.cc

```diff
@@ -1150,7 +1151,14 @@ future<> server_impl::read_barrier(seastar::abort_source* as) {
             leader = _fsm->current_leader();
         } else {
             auto applied = _applied_idx;
-            auto res = co_await get_read_idx(leader, as);
+            read_barrier_reply res;
+            try {
+                res = co_await get_read_idx(leader, as);
+            } catch (const seastar::rpc::closed_error& e) {
+                logger.trace("[{}] read_barrier on {} resulted in seastar::rpc::closed_error; retrying", _id, leader);
+                leader = server_id{};
```

Comment:
Can you insert `co_await yield()` here?

Reply:
done

can't co_await inside catch, flag is needed